tests/repository/test_upload.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import pytest
+import json
+
+from pubtools.pulplib import (
+    Repository,
+    FileRepository,
+    Task,
+    DetachedException,
+    TaskFailedException,
+)
+
+
+def test_upload_detached():
+    """publish raises if called on a detached repo"""
+    with pytest.raises(DetachedException):
+        FileRepository(id="some-repo").upload_file("some-file")
+
+
+def test_upload_file(client, requests_mocker, tmpdir, caplog):
+    """test upload a file to a repo in pulp"""
+
+    logging.getLogger().setLevel(logging.INFO)
+    caplog.set_level(logging.INFO)
+
+    repo_id = "repo1"
+    repo = FileRepository(id=repo_id)
+    repo.__dict__["_client"] = client
+
+    client._CHUNK_SIZE = 20
+
+    request_body = {
+        "_href": "/pulp/api/v2/content/uploads/cfb1fed0-752b-439e-aa68-fba68eababa3/",
+        "upload_id": "cfb1fed0-752b-439e-aa68-fba68eababa3",
+    }
+    upload_id = request_body["upload_id"]
+
+    import_report = {
+        "result": {},
+        "error": {},
+        "spawned_tasks": [
+            {
+                "_href": "/pulp/api/v2/tasks/7744e2df-39b9-46f0-bb10-feffa2f7014b/",
+                "task_id": "task1",
+            }
+        ],
+    }
+
+    tasks_report = [{"task_id": "task1", "state": "finished"}]
+
+    somefile = tmpdir.join("some-file.txt")
+    somefile.write(b"there is some binary data:\x00\x01\x02")
+
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/content/uploads/", json=request_body
+    )
+    requests_mocker.put(
+        "https://pulp.example.com/pulp/api/v2/contents/%s/0" % upload_id, json=[]
+    )
+    requests_mocker.put(
+        "https://pulp.example.com/pulp/api/v2/contents/%s/20" % upload_id, json=[]
+    )
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/repositories/%s/actions/import_upload/"
+        % repo_id,
+        json=import_report,
+    )
+    requests_mocker.delete(
+        "https://pulp.example.com/pulp/api/v2/content/uploads/%s/" % upload_id, json=[]
+    )
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/tasks/search/", json=tasks_report
+    )
+
+    # delete's done, return []
```

Comment:
It doesn't match what your docs say, which say that you'll return a "task" from deleting the upload.

Reply:
yeah, I forget changing the docs.